app/server/utils/rewards/processor.py

```diff
@@ -0,0 +1,182 @@
+from flask import g
+
+from server import db
+from server.models.user import User
+from server.models.reward import Reward
+from server.models.transfer_account import TransferAccount
+from server.models.token import Token
+from server.utils.credit_transfer import make_payment_transfer
+from server.utils.transfer_enums import TransferSubTypeEnum
+from server.utils.rewards.queries import users_with_corresponding_total_outward_txs
+
+
+def get_user_by_id(user_id: int):
+    """
+    This method queries the database for a user with an id matching the one provided as an argument
+    :param user_id: user's id
+    :return: user object
+    """
+    user = User.query.get(user_id)
+    if user:
+        return user
+    else:
+        raise Exception('User with id {} not found'.format(user_id))
+
+
+def get_transfer_account_token(transfer_account: TransferAccount):
+    """
+    Gets the token for a specific transfer account
+    :param transfer_account: transfer account object
+    :return: token object
+    """
+
+    token_id = transfer_account.token_id
+    token = Token.query.get(token_id)
+
+    if token:
+        return token
+    else:
+        raise Exception('No token found for token id {}'.format(token_id))
+
+
+def get_admin_account_to_disburse(subject_user: User):
+    """
+    This method gets all admin users for a specific organization and uses the first admin account to disburse a specific
+    amount.
+    :param subject_user:
+    :return: user object with admin role
+    """
+    # initialize organisation admins
+    organisation_admins = []
+
+    # get a user's default organisation
+    user_organisation = subject_user.default_organisation
+
+    # iterates through the organisation's users to find admins
+    for organisation_user in user_organisation.users:
+        if organisation_user.has_admin_role:
+            organisation_admins.append(organisation_user)
+
+    if len(organisation_admins) > 0:
+        # get first admin
+        admin = organisation_admins[0]
+        return admin
+    else:
+        raise ValueError('The provided user {}{} seems to have no admin users in parent organization.'.format(
+            subject_user.first_name,
+            subject_user.last_name
+        ))
+
+
+def persist_reward_data(recipients_data: dict, tag: str):
+    """
+    This methods saves reward data to the database
+    :param recipients_data:
+    :param tag:
+    """
+    reward = Reward(tag=tag)
+    db.session.add(reward)
+    reward.enter_recipient_data(recipients_data)
+    db.session.commit()
+
+
+def process_daily_bonus_query(daily_bonus_query_result: list):
```

Comment:
Perhaps this should be a temporary table?

Reply:
Could you please elaborate on this suggestion